pkg/api/agentPoolOnlyApi/vlabs/types.go

```diff
@@ -129,6 +129,7 @@ type AgentPoolProfile struct {
 	AvailabilityProfile string `json:"availabilityProfile"`
 	StorageProfile      string `json:"storageProfile" validate:"eq=StorageAccount|eq=ManagedDisks|len=0"`
 	VnetSubnetID        string `json:"vnetSubnetID,omitempty"`
+	IsOpenShiftInfra    bool   `json:"isOpenShiftInfra,omitempty"`
```

Comment:
What are the pros/cons of adding this `isOpenShiftInfra` value vs inferring via the existing `orchestratorType` property value of `"OpenShift"`? Ideally, we would not add this new typed property. (Understand there may be considerations, just would like to clarify.)

Reply:
OpenShift requires 2 types of agent nodes - one type "infra", which will run openshift value add workload (e.g. embedded registry, router, etc.), and another type (non-"infra") which runs end-user workload.  We need to distinguish these somehow and this is the switch that does it.

I suppose it could be done by the agentpoolprofile name but overloading that is a bit nasty.  I wondered whether instead of `IsOpenShiftInfra bool` a `IsInfra bool` or a `AgentRole string` (validated as an enum) might be more useful to others?

Reply:
@jim-minter's changes pushed in this PR